### DISCUSSION TITLE
[Enhancement] [select] 给select组件添加before-change事件

### DISCUSSION THREAD

**Discussion by kcmeven** [element-plus/element-plus#15019] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `select`

### 特性
1. 可以在选中时，主动控制是否选中

###  已有的例子 
如ElTabs的 before-leave

### Additional comments

希望给before-change事件，在自己点击时自己控制，是否选中
现有需求在选中时，弹窗提示，选是就选中，否就不变

**Comment by 1784123957** [comment]
很有需要
